app/routes.py

```diff
@@ -1,2 +1,333 @@
-from flask import Blueprint
+import re
+from flask import Blueprint, jsonify, request
+from flask.helpers import make_response
+from app.models.task import Task
+from app.models.goal import Goal
+import requests
+from datetime import datetime 
+from app import db
+import os
+from dotenv import load_dotenv
 
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+    
+    new_task = Task(
+        title=request_body["title"], 
+        description=request_body["description"], 
+        completed_at=request_body["completed_at"]
+        )
+    
+    db.session.add(new_task) 
+    db.session.commit()
+    
+    response_body = {
+        "task": {
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": new_task.completed_at is not None
+            }
+        }
+    
+    return jsonify(response_body), 201
+
+@tasks_bp.route("", methods=["GET"])
+def read_tasks():
+    title_query = request.args.get("sort")
+
+    if title_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif title_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all() 
+    
+    task_responses = []
+    for task in tasks:
+        task_responses.append(
+                {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at is not None
+                }
+            )
```

Comment:
This would be a good spot to use list comprehension and the helper method for the task dictionary.

```suggestion
    task_responses = [task.to_dict() for task in tasks]
```